Public/Src/Engine/UnitTests/Scheduler/PipFingerprinterTests.cs

```diff
@@ -380,6 +380,100 @@ public void TestUserProfileStaysOutOfFingerprints()
             XAssert.IsTrue(json.Contains(Path.GetFileName(Environment.GetFolderPath(Environment.SpecialFolder.InternetCache)).ToUpperInvariant()));
         }
 
+        [Fact]
+        public void TestUntrackedPathStaysOutOfFingerprints()
+        {
+            string untrackedPathText = "/x/pkgs/tool.exe";
+            var pathTable = m_context.PathTable;
+            var executable = FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, X(untrackedPathText)));
+
+            PipDataBuilder pdb = new PipDataBuilder(m_context.PathTable.StringTable);
+            string appDataSubdirName = "appdataSubDir";
+            pdb.Add(AbsolutePath.Create(pathTable, Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), appDataSubdirName)));
+            EnvironmentVariable envVar = new EnvironmentVariable(m_context.StringTable.AddString("testVar"), pdb.ToPipData(" ", PipDataFragmentEscaping.NoEscaping));
+
+            var dependencies = new HashSet<FileArtifact>() { executable };
+            var process = GetDefaultProcessBuilder(pathTable, executable)
+                .WithEnvironmentVariables(new EnvironmentVariable[] { envVar })
+                .WithDependencies(dependencies)
+                .WithUntrackedPaths(new AbsolutePath[] { AbsolutePath.Create(pathTable, X(untrackedPathText)) })
+                //.WithUntrackedScopes(new AbsolutePath[] { AbsolutePath.Create(pathTable, OperatingSystemHelper.IsUnixOS ? "/tmp" : Environment.GetFolderPath(Environment.SpecialFolder.History)) })
+                .Build();
+
+            MountPathExpander expander = new MountPathExpander(pathTable);
+
+            var fingerprinter = new PipContentFingerprinter(
+                m_context.PathTable,
+                GetContentHashLookup(executable),
+                ExtraFingerprintSalts.Default(),
+                pathExpander: expander)
+            {
+                FingerprintTextEnabled = true
+            };
+
+            var baselineFingerprint = fingerprinter.ComputeWeakFingerprint(process, out string fingerprintText);
+            fingerprintText = fingerprintText.ToUpperInvariant();
+
+            // Make sure the executable path doesn't show up in the fingerprint
+            XAssert.IsFalse(fingerprintText.Contains(executable.Path.ToString().ToUpperInvariant()));

```

Comment:
I don't know what this `fingerprintText` is supposed to contain in the first place; this text could be any kind of gibberish, so just checking that it doesn't contain executable.Path is not very convincing to me.  In fact, I just ran your test with `.WithUntrackedPaths(...)` commented out and the test still passed, meaning that indeed this is a bogus thing to check. 

In general, please change this Fact into a Theory and add a bool argument like `untrackExePath` then based on the value of that arg either (1) untrack the exe path and assert that the exe path IS NOT part of the fingerprint, or (2) don't untrack the exe path and assert that the exe path IS part of the fingerprint.

Reply:
I see now that the reason why this is is because of `executable.Path.ToString()` which always returns some bogus string; instead, use `executable.Path.ToString(pathTable)`.